cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -266,6 +271,56 @@ void EnsureLookupTablesFilled() {}
 
 #endif  // ARROW_WITH_UTF8PROC
 
+template <typename Type>
+struct AsciiReverse : StringTransform<Type, AsciiReverse<Type>> {
+  using Base = StringTransform<Type, AsciiReverse<Type>>;
+  using offset_type = typename Base::offset_type;
+
+  bool Transform(const uint8_t* input, offset_type input_string_ncodeunits,
+                 uint8_t* output, offset_type* output_written) {
+    uint8_t utf8_char_found = 0;
+    for (offset_type i = 0; i < input_string_ncodeunits; i++) {
+      // if a utf8 char is found, report to utf8_char_found
+      utf8_char_found |= input[i] & 0x80;
+      output[input_string_ncodeunits - i - 1] = input[i];
+    }
+    *output_written = input_string_ncodeunits;
+    return utf8_char_found == 0;
+  }
+
+  static Status InvalidStatus() { return Status::Invalid("Non-ascii sequence in input"); }
```

Comment:
Nit: can we be consistent with `ASCII`?

Reply:
I do not think you should add *InvalidStatus* methods with specific messages that do not necessarily generalize for all string errors. Why not directly invoke `Status::Invalid("Specific message to specific code block")`?

Reply:
I am reusing the `StringTransform` class for both ascii and utf8 kernels. In that, it always throws a `Status::Invalid("Invalid UTF8 sequence in input")` which is not quite right for an ascii kernel. That's why it was generalized and offloaded to `Derived::InvalidStatus()` :slightly_smiling_face:

Reply:
Just to clarify here, the structure of the kernel means the 'implementation' doesn't directly return an error, only true/false. Would you prefer to change it so that it returns a status instead?